### PR TITLE
Fix bone pose override not being reset when IK animation is stopped

### LINKED ISSUE #35409: Mesh/Skeleton does not return to original pose when unplay the SkeletonIK inside editor
**Godot version:** Godot 3.2 RC 2

**OS/device including version:** Windows 10

**Issue description:** Thanks for the fix for #35035.

The mesh/skeleton does not return to its original pose when unplay the SkeletonIK inside editor. It does reset when you close and reopen the scene.
bug
 
This tested in v3.2 beta 6, before the fix
expected

**Steps to reproduce:**
Open the project, play the SkeletonIKLeft inside the editor, move around the TargetRight, unplay the SkeletonIKLeft

**Minimal reproduction project:**
[ik-test.zip](https://github.com/godotengine/godot/files/4093369/ik-test.zip)

### PR DESCRIPTION
**Pull request by ericrybick**
Fixes #35409

### CODE REVIEW

**Comment by akien-mga** [comment]
Thanks!

**Comment by AndreaCatania** [comment]
@ericrybick Thanks to been working at this, however the bone reset in editor is done here: https://github.com/godotengine/godot/blob/a7f49ac9a107820a62677ee3fb49d38982a25165/editor/plugins/skeleton_ik_editor_plugin.cpp#L51-L60 a proper fix would be fix what's wrong there instead. Can you please give it a check?

**Comment by ericrybick** [comment]
Indeed, I didn't see that part.

Storing and restoring the bone_pose does nothing, because it isn't touched at all. The IK-Bones only affect the global pose override.

These are my two suggestions. The first one is more efficient and less error prone in my opinion. Not the whole skeleton is reset, just the chain items of the IK.

**Comment by AndreaCatania** [comment]
I'm really sorry but I'm able to get your suggestions, can you please rephrase your sentence?

**Comment by ericrybick** [comment]
Sorry, let me try again:

The code you've posted stores the initial bone poses in an array and restores them when the play button is not pressed anymore.

It doesn't work because IK-Bones don't change the local bone pose. IK-Bones only manipulate the `global_pose_override` for each bone in their bone-chain.

Instead of restoring the local bone poses, the pose overrides need to be disabled. This can be achieved by calling `Skeleton::set_bone_global_pose_override(...)`

Another problem is that if we continue to use `skeleton_ik_editor_plugin.cpp` for the pose reset logic, stopping the ik-playback by code will not work correctly. (The pose override will not get disabled)

I've created two different commits that do exactly that in different ways, but the second was bad so please forget about it.

So basically this PR does fix the problem in an appropriate manner, just the code you've mentioned needs to be removed.

https://github.com/ericrybick/godot/commit/ed36fb74b2279c8c72cbb255cc49f36313ea7dc5

**Comment by AndreaCatania** [comment]
This is an in editor feature and not an Inverse Kinematic feature, if the IK got changed and the plugin doesn't work properly anymore, we should expose all the functions in order to allow the plugin to reset the Skeleton back to the original position.

The editor plugins, in godot, exist to not bother the nodes with in editor features and this case is not an exception in my opinion.
